app/src/main/java/com/google/droidjump/LevelsFragment.java

```diff
@@ -17,23 +17,54 @@
 package com.google.droidjump;
 
 import static androidx.navigation.Navigation.findNavController;
+import android.annotation.SuppressLint;
 import android.os.Bundle;
+import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.Button;
-
+import android.widget.GridView;
+import android.widget.ImageButton;
+import androidx.navigation.Navigation;
 
 public class LevelsFragment extends Fragment {
 
+    private MainActivity activity;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        activity = (MainActivity) getActivity();
+    }
+
+    @SuppressLint("SetTextI18n")
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState) {
-        View rootView = inflater.inflate(R.layout.levels_screen, container, false);
-        Button levelPlayButton = rootView.findViewById(R.id.level_play);
-        levelPlayButton.setOnClickListener(view -> {
-            findNavController(view).navigate(R.id.action_levels_screen_to_game_screen);
+            Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.levels_screen, container,
+                /* attachToRoot= */ false);
+
+        // Finding gridView and putting levels to it
+        GridView gridView = rootView.findViewById(R.id.levels_grid_view);
+        int[] levels = new int[activity.getLevelsCount()];
```

Comment:
Can you move levels constructing logic in a separate method? This way it will be easier to change it later with the real implementation.

Reply:
Also, am I correct that "onCreateView" would be executed multiple times if you go to levels screen multiple times? If yes, better to move all the logic should be done only once (like getting levels) into place where it would be executed only once.

Reply:
1. Makes sense.
2. Yes, you are right. Moving it to constructor.